### PR TITLE
vesktop: 0.3.3 -> 0.4.1

### PR DESCRIPTION
**Pull request by getchoo**
## Description of changes

https://github.com/Vencord/Vesktop/compare/v0.3.3...v0.4.1

~~this was built against `nixos-unstable` to avoid building electron locally; there aren't any major changes in master as far as i can tell, so this should be fine~~ this has been built against master now :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by getchoo** [comment]
updated (and tested) to include support for [venmic](https://github.com/Vencord/venmic)

File: pkgs/by-name/ve/vesktop/package.nix [hunk 1]
```diff
@@ -92,7 +94,12 @@ stdenv.mkDerivation rec {
   # yes, upstream really packages it as "vesktop" but uses "vencorddesktop" file names
   installPhase =
     let
-      libPath = lib.makeLibraryPath [ pipewire ];
+      # this is mainly required for venmic
+      libPath = lib.makeLibraryPath [
+        libpulseaudio
+        pipewire
+        gcc13Stdenv.cc.cc.lib
+      ];
```
Comment (vgskye) [inline pkgs/by-name/ve/vesktop/package.nix:102]: mostly looks good, but does venmic have natives for aarch64? maybe vendor venmic if need be?

Comment (vgskye) [reply pkgs/by-name/ve/vesktop/package.nix:102]: actually, aarch64 is already broken so it's not hurting much of anything. aarch64 being (possibly) broken could be addressed in a future pr

**Review by vgskye — APPROVED** [review]

File: pkgs/by-name/ve/vesktop/package.nix (hunk 1 above)
Comment (getchoo) [reply pkgs/by-name/ve/vesktop/package.nix:102]: i think a future pr would be a good idea. i'll change the platforms and broken status for this pr until we can get everything resolved

**Review by vgskye — APPROVED** [review]

**Comment by pyrox0** [comment]
LGTM! I think 0.4.1 has just been released, so I'd update to that before merging. Looks good otherwise!

**Review by pyrox0 — APPROVED** [review]

**Comment by getchoo** [comment]
> I think 0.4.1 has just been released

thanks for bringing it up! this pr is updating to 0.4.1 now :)

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 262652` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vesktop</li>
  </ul>
</details>

**Review by vgskye — APPROVED** [review]
gee, ofborg, do you think I'll approve the exact same thing I approved before just with the version number and source hashes changed?

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**